docs/source/opener_parameters_api.md

```diff
@@ -0,0 +1,220 @@
+be represented fully by splitting datasets in this way. In these cases:
+
+  1. The JSON Schema SHOULD describe the smallest possible superset of the
+     allowed parameter combinations.
+
+  2. The additional restrictions on parameter combinations MUST be clearly
+     documented.
+
+  3. If illegal parameter combinations are supplied, the opener MUST raise an
+     exception with an informative error message, and the user interface
+     should present this message clearly to the user.
+
+## Common parameters
+
+While an opener is free to define any opener parameters for any of its
+datasets, there are some common parameters which are likely to be used by the
+majority of datasets. Any dataset requiring the specification of these
+parameters MUST use the standard parameter names, syntax, and semantics
+defined below, in order to keep the interface consistent. For instance, if a
+dataset allows a time aggregation period to be specified, it MUST use the
+`time_period` parameter with the format described below rather than some other
+alternative name and/or format. The parameters are described below with their
+Python type annotations.
+
+ - `variable_names: List[str]`  
+   A list of the identifiers of the requested variables.
+ - `bbox: Union[str,Tuple[float, float, float, float]]`
+   The bounding box for the requested data, in the order x0, y0, x1, y1.
+ - `crs: str`  
+   The identifier for the co-ordinate reference system of geographic data.
+ - `spatial_res: float`  
+   The requested spatial resolution (x and y) of the returned data.
+ - `time_range: Tuple[Optional[str], Optional[str]]`  
+   The requested time range for the data to be returned. See section
+   ‘Date, time, and duration specifications’ below.
+ - `time_period: str`  
+   The requested temporal aggregation period for the data. See section
+   ‘Date, time, and duration specifications’ below.
+
+## Semantics of list-valued parameters
+
+The `variables` parameter takes as its value a list, with no members repeated
+and the values of its members drawn from a predefined set. The values of this
+parameter, and other parameters whose values also follow such a format, are
+interpreted by xcube as a *restriction*, much like a bounding box or time
+range. That is:
+
+ - By default (if the parameter is omitted or if a `None` value is supplied
+   for it), *all* the possible member values MUST be included in the list. In
+   the case of `variables`, this will result in a dataset containing all the
+   available variables.
+ - If a list containing *some* of the possible members is given, a dataset
+   corresponding to those members only MUST be returned. In the case of
+   `variables`, this will result in a dataset containing only the requested
+   variables.
+ - As a special case of the above, if an empty list is supplied, a dataset
+   containing *no data* MUST be returned -- but with the requested spatial and
+   temporal dimensions.
+
+## Date, time, and duration specifications
```

Comment:
If I understand correctly time periods are specified only by a duration character e.g. M or Y. However, they are in the same chapter of the date/time RFC you quoted.

Reply:
Right!

Reply:
I deliberately left this vague, and should have marked it explicitly as a TODO. We should discuss exactly what we want to support as a period specification -- I couldn't find it detailed anywhere in the xcube docs.

Reply:
As to the idea that month units are forbidden for time periods in pandas or xarray, I can't find any evidence of this. For instance, [here's](http://xarray.pydata.org/en/stable/examples/weather-data.html?highlight=month#Monthly-averaging) an example of monthly averaging with xarray and [here's](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timeseries-offset-aliases) the extensive list of pandas time series frequency specifiers (which includes monthly). I'm not sure whether we'd want to explicitly support the powerful and complex pandas schema; perhaps a small subset would be safer.

Reply:
I suggest that, for now, we define the time period format as [integer][unit], where [unit] is one of HDWMY (hour, day, week, month, year), and [integer] may be omitted to indicate 1. This could even be validated by a simple JSON Schema format regex.

Reply:
Suggestion in previous comment now implemented.